### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by JudithRe**
As per [US13](https://github.com/JudithRe/capstone-flashcard-dictionary/issues/24)
- created login
   - if not login, will always be redirected to login page. 
   - for now, all pages are visible as soon as logged in. 
- created User Model for MongoDB
   - passwords are encrypted

### CODE REVIEW

File: components/AuthForm/index.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+function AuthForm() {
+  const [registered, setRegistered] = useState(false);
+  const usernameInputRef = useRef();
+  const passwordInputRef = useRef();
+  const [loginError, setLoginError] = useState({
+    hasError: false,
+    errorMessage: "",
+  });
+
+  const [isLogin, setIsLogin] = useState(true);
```
Comment (CJAkkuec) [inline components/AuthForm/index.js]: If I'm understanding what's happening correctly, then I'd probably prefer the naming here to reflect what `isLogin` does - something like `loginProcess` or similar. As it stands, `isLogin` is a little bit difficult to predict, which makes the code more difficult to understand.

File: components/HeaderBackground/index.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ const StyledCircle = styled.div`
   height: 300px;
   border-radius: 50%;
   background-color: var(--highlight-red);
+  color: var(--white);
```
Comment (CJAkkuec) [inline components/HeaderBackground/index.js]: I'd probably argue that `--white` is not really a variable (or should be one). If it's just `#fff` then perhaps describing what that white is used for would be more apt.

File: components/Signout/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      </>
+    );
+  }
+}
+
+const InlineParagraph = styled.p`
+  display: inline-block;
+  background-color: transparent;
+  margin-right: 10px;
+`;
```
Comment (CJAkkuec) [inline components/Signout/index.js]: Why not use `<span>`? That's more or less a default inline paragraph. And I'm not sure if `background-color: transparent;` needs to be defined.

Edit: Not actually sure if the above is the right call in this case, but it's worth testing out.

File: pages/add/index.js [hunk 4]
```diff
@@ -3,13 +3,33 @@ import EntriesContainer from "@/components/EntriesContainer";
...
+}
 
 export default function AddEntries({
   handleAddEntry,
   wordList,
   setActivePage,
 }) {
-  setActivePage("add");
+  useEffect(() => {
+    setActivePage("add");
```
Comment (CJAkkuec) [inline pages/add/index.js]: Usually we'd rather not pass the setters themselves to the components, but rather a handler function that's defined where the setter is initialized.

File: pages/words/index.js [hunk 5]
```diff
@@ -14,43 +30,48 @@ export default function WordList({
   setIsDetailEditMode,
   setActivePage,
```
Comment (CJAkkuec) [inline pages/words/index.js]: The same thing about setters and handlers applies here.

**Review by CJAkkuec — CHANGES_REQUESTED** [review]
Looking good overall,  just a couple of small notes!

File: pages/add/index.js (hunk 4 above)
Comment (JudithRe) [reply pages/add/index.js]: Oh, okay. So would I just write a function that just includes the setter then? 

like 
```JS
function activePage(value) {
setActivePage(value)}
```
?

Comment (CJAkkuec) [reply pages/add/index.js]: Probably something like `handleActivePage`, but yeah. :)

Comment (JudithRe) [reply pages/add/index.js]: Okay. Thank you. I will add that :)

**Pull request merged**